src/Stacktrace.php

```diff
@@ -145,14 +145,26 @@ public function addFrame(string $file, int $line, array $backtraceFrame): void
 
         if (null !== $this->options->getProjectRoot()) {
             $excludedAppPaths = $this->options->getInAppExcludedPaths();
+            $includedAppPaths = $this->options->getInAppIncludedPaths();
             $absoluteFilePath = @realpath($file) ?: $file;
-            $isApplicationFile = 0 === strpos($absoluteFilePath, $this->options->getProjectRoot());
+            $isApplicationFile = 0 === strpos(
+                $absoluteFilePath, $this->options->getProjectRoot()
+            );
 
             if ($isApplicationFile && !empty($excludedAppPaths)) {
-                foreach ($excludedAppPaths as $path) {
-                    if (0 === mb_strpos($absoluteFilePath, $path)) {
-                        $frame->setIsInApp(false);
-
+                foreach ($excludedAppPaths as $excludedPath) {
```

Comment:
This loop seems a bit convoluted, we could take the occasion to refactor it and/or extract to a private method.
Also, it seems we're not handling the case of a in-app path that is not under the project root, am I right? @HazAT what's the expected behavior here?

Reply:
What am I supposed to do here?

Reply:
I might be wrong, but I cannot find in the documentation any hint to the `project_root` option, so it may be something that we only have in the PHP SDK and that we may decide to deprecate. Thinking that all the application source code is under just one folder may be a too strong assumption, e.g. in Symfony `2.x` the `AppKernel` class was in a folder different than the one that contained the rest of the source code but technically speaking is part of the application's source. With the `include_app_paths` and the `exclude_app_paths` the user has full flexibility in regards to defining what belongs or not to the application and doesn't need a third option just to restrict to which is the base path those options apply to

Reply:
Still, we cannot change this behavior without breaking BC. I would prefer to deprecate/change that in a different PR.

I would leave this piece like this for now, and refactor it later.